igvc_navigation/src/path_follower/Path_follower.cpp

```diff
@@ -1,37 +1,64 @@
 #define _USE_MATH_DEFINES
 
+#include <Eigen/Dense>
+#include <cmath>
+#include <iostream>
+
 #include <geometry_msgs/PointStamped.h>
 #include <geometry_msgs/PoseStamped.h>
 #include <igvc_msgs/velocity_pair.h>
 #include <nav_msgs/Odometry.h>
-#include <nav_msgs/Path.h>
-#include <ros/ros.h>
+
 #include <tf/transform_datatypes.h>
-#include <Eigen/Dense>
-#include <cmath>
+
 #include <igvc_utils/NodeUtils.hpp>
 #include <igvc_utils/RobotState.hpp>
-#include <iostream>
-#include "SmoothControl.h"
 
-ros::Publisher cmd_pub;
-ros::Publisher target_pub;
-ros::Publisher trajectory_pub;
+#include "Path_follower.h"
 
-nav_msgs::PathConstPtr path;
-geometry_msgs::PointStampedConstPtr waypoint;
+Path_follower::Path_follower()
+{
+  ros::NodeHandle nh;
+  ros::NodeHandle pNh("~");
+
+  // load controller parameters
+  double target_velocity;
+  double axle_length;
+  double k1;
+  double k2;
+  double granularity;
+  double lookahead_dist;
+  igvc::param(pNh, "target_v", target_velocity, 1.0);
+  igvc::param(pNh, "axle_length", axle_length, 0.52);
+  igvc::param(pNh, "k1", k1, 1.0);
+  igvc::param(pNh, "k2", k2, 3.0);
+  igvc::param(pNh, "granularity", granularity, 2.0);
+  igvc::param(pNh, "lookahead_dist", lookahead_dist, 2.0);
+  controller = std::unique_ptr<Smooth_control>(
+      new Smooth_control{ k1, k2, axle_length, granularity, target_velocity, lookahead_dist });
+
+  // load global parameters
+  igvc::param(pNh, "maximum_vel", maximum_vel, 1.6);
```

Comment:
I think it would be safer to make `maximum_vel` a required launch parameter. Maybe @JasonGibson274 can weigh in but the maximum velocity is definitely something we want to be aware of.